airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java

```diff
@@ -40,7 +42,7 @@ public static MongoClient createMongoClient(final MongoDbSourceConfig config) {
     if (config.hasAuthCredentials()) {
       final String authSource = config.getAuthSource();
       final String user = config.getUsername();
```

Comment:
Does user need encoding as well?

Reply:
[It appears so](https://www.mongodb.com/docs/manual/reference/connection-string/). I didn't realize usernames were allowed to have special characters in the first place.

Reply:
It doesn't look like any of the class that generate the client connection string take care of encoding. 
great stuff!

Reply:
Thanks, just addressed the feedback!